baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py

```diff
@@ -56,11 +58,48 @@ class HPSNMPProvisioningDriver(api.NetworkProvisioningApi):
 
     def __init__(self):
         """initialize the snmp driver."""
+        self.conf = cfg.CONF
+        self.bnp_sync_enable = self.conf.default.bnp_sync_enable
+        self.bnp_sync_interval = float(
+            self.conf.default.get('bnp_sync_interval'))
+        if self.bnp_sync_enable:
+            self.start_snmp_polling()
         # TODO(selva) need to check how we can load dynamically
         drvr = 'baremetal_network_provisioning.drivers.snmp_driver.SNMPDriver'
         self.context = neutron_context.get_admin_context()
         self._load_drivers(drvr)
 
+    def _snmp_sync_thread(self):
+        """Sync switch database periodically."""
+        LOG.info(_LI('BMNP start thread....'))
+        while True:
+            portmaps = db.get_all_bnp_swport_mappings(self.context)
+            for portmap in portmaps:
+                swport = db.get_bnp_phys_switch_port_by_id(
+                    self.context, portmap['switch_port_id'])
+                switch = db.get_bnp_phys_switch(self.context,
+                                                portmap['switch_id'])
+                try:
+                    snmp_drv = discovery_driver.SNMPDiscoveryDriver(switch)
```

Comment:
This code is not working please resolve the below error ,
 SNMP operation 'GET' failed: NoSuchObjectError({'str': "Can't resolve node name ::(u'1', u'3', u'6', u'1', u'2', u'1', u'2', u'2', u'1', u'8', u'12') at <pysnmp.smi.view.MibViewController instance at 0x7f1e0c2dad40>"})^[[00m